### PR TITLE
rearrange directories

### PR DESCRIPTION
**Pull request by simonbyrne**
- move tests to tests
- get rid of intermediate src directories

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -33,6 +33,7 @@ gpu:test:dev:
   extends: .gputest
   variables:
     CI_VERSION_TAG: 'dev'
+  allow_failure: true
```
Comment (jkozdon) [inline .gitlab-ci.yml]: Just an FYI that this means we are effectively turning off GPU CI for the balance law solver.

Comment (charleskawczynski) [reply .gitlab-ci.yml]: @simonbyrne why was this necessary?

Comment (simonbyrne) [reply .gitlab-ci.yml]: See #232

**Pull request merged**